cura/CuraApplication.py

```diff
@@ -1382,22 +1383,65 @@ def reloadAll(self) -> None:
         if not nodes:
             return
 
+        objects_in_filename = {}  # type: Dict[str, List[CuraSceneNode]]
         for node in nodes:
             mesh_data = node.getMeshData()
-
             if mesh_data:
                 file_name = mesh_data.getFileName()
                 if file_name:
-                    job = ReadMeshJob(file_name)
-                    job._node = node  # type: ignore
-                    job.finished.connect(self._reloadMeshFinished)
-                    if has_merged_nodes:
-                        job.finished.connect(self.updateOriginOfMergedMeshes)
-
-                    job.start()
+                    if file_name not in objects_in_filename:
+                        objects_in_filename[file_name] = []
+                    if file_name in objects_in_filename:
+                        objects_in_filename[file_name].append(node)
                 else:
                     Logger.log("w", "Unable to reload data because we don't have a filename.")
 
+        for file_name, nodes in objects_in_filename.items():
+            for object_index, node in enumerate(nodes):
+                job = ReadMeshJob(file_name)
+                job.object_to_be_reloaded = self._getObjectIndexInFile(file_name, node.getName())  # The object index in file to be loaded by this specific job
+                job._node = node  # type: ignore
+                job.finished.connect(self._reloadMeshFinished)
+                if has_merged_nodes:
+                    job.finished.connect(self.updateOriginOfMergedMeshes)
+
+                job.start()
+
+    @staticmethod
+    def _getObjectIndexInFile(file_name: str, node_name: str) -> int:
```

Comment:
I'm not really a big fan of this solution. Right now the name is generated from the objectId, but it's not really a future proof way of checking for it.
Soo, instead of that, I think it's better to add an objectID property that can be used to do this. That way, we keep open the ability to let the user change the names of objects.